pkg/macos/pkg-scripts/preinstall

```diff
@@ -96,12 +96,20 @@ if [ -L "$SBIN_DIR/salt-config" ]; then
 fi
 
 #-------------------------------------------------------------------------------
-# Remove the $INSTALL_DIR directory
+# Remove folders and files from the INSTALL_DIR
+# Don't remove extras-3.10
 #-------------------------------------------------------------------------------
-if [ -d "$INSTALL_DIR" ]; then
-    log "Cleanup: Removing $INSTALL_DIR"
-    rm -rf "$INSTALL_DIR"
-    log "Cleanup: Removed Successfully"
+for dir in "$INSTALL_DIR"/*/; do
+    if [[ "$dir" != *"extras-3."* ]]; then
+        log "Cleanup: Removing $dir"
+        rm -rf "$dir"
+    fi
+done
```

Comment:
Do we want to target specifically the extras directory related to the python version of the currently installed version of salt?   For instance, if we upgrade to 3.11 in a future release, this will keep extras-3.10 around and still add an extras-3.11

Reply:
I can do that

Reply:
OK, the Python version is dynamically updated when the package is built depending on the version of Python that this installed via relenv.